cpp/dolfinx/fem/interpolate.h

```diff
@@ -598,8 +629,252 @@ void interpolate(Function<T>& u, const Function<T>& v,
     // Get mesh and check that functions share the same mesh
     if (mesh != v.function_space()->mesh())
     {
-      throw std::runtime_error(
-          "Interpolation on different meshes not supported (yet).");
+      const int nProcs = dolfinx::MPI::size(MPI_COMM_WORLD);
+      const auto mpi_rank = dolfinx::MPI::rank(MPI_COMM_WORLD);
+
+      const int tdim = u.function_space()->mesh()->topology().dim();
+      const auto cell_map
+          = u.function_space()->mesh()->topology().index_map(tdim);
+      const int num_cells = cell_map->size_local() + cell_map->num_ghosts();
+      std::vector<std::int32_t> cells(num_cells, 0);
+      std::iota(cells.begin(), cells.end(), 0);
+
+      // Collect all the points at which values are needed to define the
+      // interpolating function
+      const xt::xtensor<double, 2> x = fem::interpolation_coords(
+          *u.function_space()->element(), *u.function_space()->mesh(), cells);
+
+      // This transposition is a quick and dirty solution and should be avoided
+      auto x_t = xt::zeros_like(xt::transpose(x));
+      for (decltype(x.shape(1)) i = 0; i < x.shape(1); ++i)
+      {
+        for (decltype(x.shape(0)) j = 0; j < x.shape(0); ++j)
+        {
+          x_t(i, j) = x(j, i);
+        }
+      }
+
+      // Create a global bounding-box tree to quickly look for processes that
+      // might be able to evaluate the interpolating function at a given point
+      dolfinx::geometry::BoundingBoxTree bb(*v.function_space()->mesh(), tdim,
+                                            0.0001);
+      auto globalBB = bb.create_global_tree(MPI_COMM_WORLD);
```

Comment:
I would use `mesh->comm()` here to be on the safe side (in case mesh is defined with `MPI_COMM_SELF`). Do you agree?

Reply:
Yes, of course. Good catch.